### PR TITLE
Add support for Http rule additional_bindings

### PR DESCRIPTION
**Pull request by qiwzhang**
https://github.com/cloudendpoints/esp/issues/364

### CODE REVIEW

File: src/api_manager/config_test.cc [hunk 1]
```diff
@@ -900,6 +900,36 @@ TEST(Config, TestHttpOptions) {
...
 
+TEST(Config, TestAdditionalBindings) {
+  MockApiManagerEnvironmentWithLog env;
+
+  static const char config_text[] = R"(
+ name: "Service.Name"
+ http {
+   rules {
+     selector: "GetMessages"
+     get: "/v1/messages/{message_id}"
```
Comment (TAOXUY) [inline src/api_manager/config_test.cc]: add test covering the case of failed primary httpRule with additional_rules

Comment (qiwzhang) [reply src/api_manager/config_test.cc]: Done

File: src/api_manager/config.h [hunk 2]
```diff
@@ -106,10 +106,16 @@ class Config {
   bool LoadService(ApiManagerEnvInterface *env,
                    const std::string &service_config);
 
-  // Create MethodInfo for HTTP methods, register them to PathMatcher.
+  // Loads the Http rules, registers and adds CORS support if required.
   bool LoadHttpMethods(ApiManagerEnvInterface *env,
                        PathMatcherBuilder<MethodInfo *> *pmb);
 
+  // Register a Http rule to the path matcher builder.
```
Comment (TAOXUY) [inline src/api_manager/config.h]: comment OPTION method is excluded

File: src/api_manager/config_test.cc [hunk 3]
```diff
@@ -900,6 +900,92 @@ TEST(Config, TestHttpOptions) {
...
+
+  auto method = config->GetMethodInfo("GET", "/v1/messages/1234");
+  ASSERT_NE(nullptr, method);
+
+  ASSERT_EQ(method,
+            config->GetMethodInfo("GET", "/v1/users/123/messages/1234"));
+  ASSERT_EQ(method, config->GetMethodInfo("GET", "/v2/messages/123/1234"));
+}
+
+TEST(Config, TestAdditionalBindingsFailedTopLevel) {
```
Comment (TAOXUY) [inline src/api_manager/config_test.cc]: Added test for allow_cors and not_allow_cors

Comment (qiwzhang) [reply src/api_manager/config_test.cc]: Done

File: src/api_manager/config.h (hunk 2 above)
Comment (qiwzhang) [reply src/api_manager/config.h]: No need to add here

**Pull request merged**